src/sqlfluff/core/linter/common.py

```diff
@@ -34,25 +39,86 @@ class RenderedFile(NamedTuple):
     source_str: str
 
 
+class ParsedVariant(NamedTuple):
+    """An object to store the result of parsing a single TemplatedFile.
+
+    Args:
+        templated_file (:obj:`TemplatedFile`): Containing the details
+            of the templated file. If templating fails, this will be `None`.
+        tree (:obj:`BaseSegment`): The segment structure representing the
+            parsed file. If parsing fails due to an unrecoverable
+            violation then we will be None.
+        lexing_violations (:obj:`list` of :obj:`SQLLexError`): Any violations
+            raised during the lexing phase.
+        parsing_violations (:obj:`list` of :obj:`SQLParseError`): Any violations
+            raised during the lexing phase.
+    """
+
+    templated_file: Optional[TemplatedFile]
+    tree: Optional[BaseSegment]
+    lexing_violations: List[SQLLexError]
+    parsing_violations: List[SQLParseError]
+
+    def violations(self) -> List[Union[SQLLexError, SQLParseError]]:
+        """Returns the combined set of violations for this variant."""
```

Comment:
```suggestion
        """Returns the combined lexing and parsing violations for this variant."""
```
Small edit - I know lexing and parsing are necessarily separate - but it's not a set, and at the time of writing, parsing violations are not deduped between variants